src/main/io/serial_1wire_vcp.c

```diff
@@ -0,0 +1,237 @@
+#include "drivers/serial_usb_vcp.h"
+
+uint8_t escCount;
+
+static escHardware_t escHardware[MAX_PWM_MOTORS];
+
+static uint32_t GetPinPos(uint32_t pin) {
+    uint32_t pinPos;
+    for (pinPos = 0; pinPos < 16; pinPos++) {
+        uint32_t pinMask = (0x1 << pinPos);
+        if (pin & pinMask) {
+            return pinPos;
+        }
+    }
+    return 0;
+}
+static uint8_t selected_esc;
+
+#define ESC_IS_HI   digitalIn(escHardware[selected_esc].gpio, escHardware[selected_esc].pin) != Bit_RESET
+#define ESC_IS_LO   digitalIn(escHardware[selected_esc].gpio, escHardware[selected_esc].pin) == Bit_RESET
+#define ESC_SET_HI  digitalHi(escHardware[selected_esc].gpio, escHardware[selected_esc].pin)
+#define ESC_SET_LO  digitalLo(escHardware[selected_esc].gpio, escHardware[selected_esc].pin)
+
+#define ESC_INPUT   gpioInit(escHardware[selected_esc].gpio, &escHardware[selected_esc].gpio_config_INPUT)
+#define ESC_OUTPUT  gpioInit(escHardware[selected_esc].gpio, &escHardware[selected_esc].gpio_config_OUTPUT)
+
+void usb1WireInitializeVcp(void)
+{
+    pwmDisableMotors();
+    selected_esc = 0;
+    memset(&escHardware, 0, sizeof(escHardware));
+    pwmIOConfiguration_t *pwmIOConfiguration = pwmGetOutputConfiguration();
+    for (volatile uint8_t i = 0; i < pwmIOConfiguration->ioCount; i++) {
+      if ((pwmIOConfiguration->ioConfigurations[i].flags & PWM_PF_MOTOR) == PWM_PF_MOTOR) {
+          if(motor[pwmIOConfiguration->ioConfigurations[i].index] > 0) {
+              escHardware[selected_esc].gpio = pwmIOConfiguration->ioConfigurations[i].timerHardware->gpio;
+              escHardware[selected_esc].pin = pwmIOConfiguration->ioConfigurations[i].timerHardware->pin;
+              escHardware[selected_esc].pinpos = GetPinPos(escHardware[selected_esc].pin);
+              escHardware[selected_esc].gpio_config_INPUT.pin = escHardware[selected_esc].pin;
+              escHardware[selected_esc].gpio_config_INPUT.speed = Speed_2MHz; // see pwmOutConfig()
+              escHardware[selected_esc].gpio_config_INPUT.mode = Mode_IPU;
+              escHardware[selected_esc].gpio_config_OUTPUT = escHardware[selected_esc].gpio_config_INPUT;
+              escHardware[selected_esc].gpio_config_OUTPUT.mode = Mode_Out_PP;
+              ESC_INPUT;
+              ESC_SET_HI;
+              selected_esc++;
+          }
+      }
+    }
+    escCount = selected_esc;
+    selected_esc = 0;
+}
+
+void usb1WireDeInitializeVcp(void){
+    for (selected_esc = 0; selected_esc < (escCount); selected_esc++) {
+        escHardware[selected_esc].gpio_config_OUTPUT.mode = Mode_AF_PP; // see pwmOutConfig()
+        ESC_OUTPUT;
+        ESC_SET_LO;
+    }
+    escCount = 0;
+    pwmEnableMotors();
+}
+
+#define START_BIT_TIMEOUT_MS 2
+
+#define BIT_TIME (52)  //52uS
+#define BIT_TIME_HALVE (BIT_TIME >> 1) //26uS
+#define START_BIT_TIME (BIT_TIME_HALVE + 1)
+#define STOP_BIT_TIME ((BIT_TIME * 9) + BIT_TIME_HALVE)
+
+static void suart_putc_(uint8_t tx_b)
```

Comment:
```
uint16_t bitmask = (tx_b << 1) | (1 << 9);
```
now just shift out bits (including startbit and stopbit) until bitmask is zero (stopbit shifted out)